### ISSUE TITLE
Exception when Connection's _handler becomes null if MariaDB 10.2 is cleanly stopped

### ISSUE THREAD

**Issue by hoylen** [dart-db/sqljocky5#8]
When the _mysqld_ server is cleanly shutdown and a client connection pool has connections open, the client crashes because `_handler` becomes null. Line 213 of _connection.dart_ tries to invoke the _processResponse_ method on the null.

This problem occurs when MariaDB 10.2 is used as the _mysqld_ server. It didn't occur before with MariaDB 5.5.56.

By cleanly shutdown, I mean it was stopped by running either "systemctl stop mariadb.service" or "systemctl restart mariadb.service". If you stop _mysqld_ with a "kill -9" and then start _mysqld_ up again (or even by rebooting the host machine) before the client's next SQL query, then it works fine (no exception is raised, and the connection pool automatically re-establishes a new connection to process the SQL query).

I suspect MariaDB 10.2 is now doing something different when it shuts down "cleanly" and _sqljocky5_ isn't expecting that.

Note: if the _wait_timeout_ has been set to a low value, so connections have timed out, then cleanly shutting down _mysqld_ does not cause this problem. It seems to only happen when the connection pool is still holding active/open connections.

```
Unhandled exception:
NoSuchMethodError: The method 'processResponse' was called on null.
Receiver: null
Tried calling: processResponse(Instance of 'Buffer')
#0      Connection._handleData (package:sqljocky5/src/connection.dart:247:9)
<asynchronous suspension>
#1      Connection._handleHeader (package:sqljocky5/src/connection.dart:180:18)
<asynchronous suspension>
#2      Connection.readPacket (package:sqljocky5/src/connection.dart:165:7)
<asynchronous suspension>
#3      BufferedSocket._onData (package:sqljocky5/src/buffered_socket.dart:92:22)
#4      _RootZone.runUnaryGuarded (dart:async/zone.dart:1307)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:330)
#6      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:257)
#7      _StreamController&&_SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:796)
#8      _StreamController._add (dart:async/stream_controller.dart:667)
#9      _StreamController.add (dart:async/stream_controller.dart:613)
#10     new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1203)
#11     _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:760)
#12     _microtaskLoop (dart:async/schedule_microtask.dart:41)
#13     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#14     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:99)
#15     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:152)

Process finished with exit code 255
```

Line 206-213 of _connection.dart_:
```dart
 _handleData(buffer) async {
    _readyForHeader = true;
    //log.fine("read all data: ${_dataBuffer._list}");
    //log.fine("read all data: ${Buffer.listChars(_dataBuffer._list)}");
    _headerBuffer.reset();

    try {
      var response = _handler.processResponse(buffer);
```

**Comment by hoylen** [comment]
This issue seems to have been solved in sqljocky v2.2.1.

Testing with <https://github.com/hoylen/sqljocky5-recovery> shows that sqljocky v2.2.1 does not show this behaviour anymore.

**Comment by tejainece** [comment]
Thanks for verifying it!
